crates/sncast/src/starknet_commands/utils/serialize.rs

```diff
@@ -0,0 +1,131 @@
+use anyhow::{Context, Result, bail};
+use camino::Utf8PathBuf;
+use clap::Args;
+use data_transformer::transform;
+use foundry_ui::UI;
+use sncast::{
+    get_class_hash_by_address, get_contract_class,
+    helpers::{configuration::CastConfig, rpc::RpcArgs},
+    response::{errors::StarknetCommandError, serialize::SerializeResponse},
+};
+use starknet::core::{
+    types::{ContractClass, contract::AbiEntry},
+    utils::get_selector_from_name,
+};
+use starknet_types_core::felt::Felt;
+
+#[derive(Args, Clone, Debug)]
+#[group(required = true, multiple = false)]
+pub struct LocationArgs {
+    /// Class hash of contract which contains the function
+    #[arg(short = 'c', long)]
+    pub class_hash: Option<Felt>,
+
+    /// Address of contract which contains the function
+    #[arg(short = 'd', long)]
+    pub contract_address: Option<Felt>,
+
+    /// Path to the file containing ABI of the contract class
+    #[arg(long)]
+    pub abi_file: Option<Utf8PathBuf>,
+}
+
+#[derive(Debug)]
+pub enum Location {
+    AbiFile(Utf8PathBuf),
+    ClassHash(Felt),
+    ContractAddress(Felt),
+}
+
+impl TryFrom<LocationArgs> for Location {
+    type Error = anyhow::Error;
+
+    fn try_from(args: LocationArgs) -> Result<Self> {
+        match (args.class_hash, args.contract_address, args.abi_file) {
+            (Some(class_hash), None, None) => Ok(Location::ClassHash(class_hash)),
+            (None, Some(address), None) => Ok(Location::ContractAddress(address)),
+            (None, None, Some(path)) => Ok(Location::AbiFile(path)),
+            _ => bail!(
+                "Exactly one of --class-hash, --contract-address, or --abi-file must be provided"
+            ),
+        }
+    }
+}
+
+#[derive(Args, Clone, Debug)]
+#[command(about = "Serialize Cairo expressions into calldata")]
+pub struct Serialize {
+    /// Comma-separated string of Cairo expressions
+    #[arg(long, allow_hyphen_values = true)]
+    pub arguments: String,
+
+    /// Name of the function whose calldata should be serialized
+    #[arg(short, long)]
+    pub function: String,
+
+    #[command(flatten)]
+    pub location_args: LocationArgs,
+
+    #[command(flatten)]
+    pub rpc_args: Option<RpcArgs>,
```

Comment:
I'd manually verify if `Option` behaves as intended with command flatten. I remember having some issues with that though don't remember the details.

Reply:
Though I see that you added tests for these which pass, so it might be non issue

Reply:
Yup, I've covered it.